sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClientBase.java

```diff
@@ -86,7 +86,7 @@ public abstract class IdentityClientBase {
     static final String WINDOWS_SWITCHER = "/c";
     static final String LINUX_MAC_SWITCHER = "-c";
     static final Pattern WINDOWS_PROCESS_ERROR_MESSAGE = Pattern.compile("'azd?' is not recognized");
-    static final Pattern SH_PROCESS_ERROR_MESSAGE = Pattern.compile("azd?: command not found");
+    static final Pattern SH_PROCESS_ERROR_MESSAGE = Pattern.compile("azd?:.*not found");
```

Comment:
Sort of, but they aren't great either. Effectively we'd be either calling or implementing `which`/`where`. `cmd.exe` and `sh` are very unlikely to ever change their outputs, and since we're specifically calling them we aren't going to get duped into running another terminal.

For example, just trying to launch `az`:
```
jshell> var p = new ProcessBuilder("az", "--help").inheritIO()
p ==> java.lang.ProcessBuilder@6fdb1f78

jshell> p.start()
|  Exception java.io.IOException: Cannot run program "az": CreateProcess error=2, The system cannot find the file specified
|        at ProcessBuilder.start (ProcessBuilder.java:1143)
|        at ProcessBuilder.start (ProcessBuilder.java:1073)
|        at (#12:1)
|  Caused by: java.io.IOException: CreateProcess error=2, The system cannot find the file specified
|        at ProcessImpl.create (Native Method)
|        at ProcessImpl.<init> (ProcessImpl.java:499)
|        at ProcessImpl.start (ProcessImpl.java:158)
|        at ProcessBuilder.start (ProcessBuilder.java:1110)
|        ...
```

We could change the callsite to have a platform specific command line (though I haven't tried it on Mac yet), which would work.

That all said, I think this approach is fine and has worked for a long time. I just broke it when I made a fix a few weeks ago.

If you feel strongly I'll figure something else out.

Reply:
I have no strong feelings, just wanted to make sure we weren't going to get ourselves stuck at some point having to fix this again in the future, and based on your response that seems like a no.